catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/security/WorkspacePolicyExtensionTest.java

```diff
@@ -102,131 +103,177 @@ public int hashCode() {
         };
     }
 
-    @Test
-    public void testShouldKeepAllWhenNoneImplied() {
-        List<Permission> before = ImmutableList.of(RANDOM, ROLES, EMAILS);
-
-        doReturn(before).when(match)
+    private static KeyValueCollectionPermission coll(List<Permission> permissions) {
+        KeyValueCollectionPermission match = mock(KeyValueCollectionPermission.class);
+        doReturn(permissions).when(match)
                 .getPermissionList();
+        return match;
+    }
+
+    @Test
+    public void testWorkspaceTagShouldAlwaysBeImplied() {
+        List<Permission> before = ImmutableList.of(WORKSPACE, RANDOM);
 
         CollectionPermission subject = makeSubject((p) -> false);
 
-        List<Permission> after = extension.isPermittedMatchAll(subject, match)
+        List<Permission> after = extension.isPermittedMatchAll(subject, coll(before))
                 .getPermissionList();
 
-        assertThat(after, is(before));
+        assertThat(after, is(ImmutableList.of(RANDOM)));
```

Comment:
❓ Why does this need to be wrapped in an immutable list?

Reply:
It needs to be a list because `RANDOM` is just a permission, not a list of random permissions and the policy extension returns a list of permissions.